docs/src/adr/008adr-exceptions.md

```diff
@@ -0,0 +1,122 @@
+# ADR-008: Apalache Exceptions
+
+| author     | revision |
+| ------------ | --------:|
+| Jure Kukovec |    1 |
+
+This ADR documents the various exception thrown in Apalache, and the circumstances that trigger them.
+
+## 1. User input exceptions
+Exceptions in this family are caused by incorrect input from the user. All of these exceptions should exit cleanly and should NOT report a stack trace.
```

Comment:
Should we use [functional error handling](https://docs.scala-lang.org/overviews/scala-book/functional-error-handling.html) for these kinds of errors, or perhaps `Either` types for these kinds of errors?

This can ensure we don't every leave a non-fatal exception unhandled.

Reply:
Sure, but that's more of an implementation detail than design, I'd say.

Reply:
Is this more of an implementation detail than what is proposed in Section 3? Perhaps the design-level constraint to add here is: "We should be able to statically enforce that none of these exceptions can be left unhandled"?

Reply:
SG, I'll add that.